### PR TITLE
Slashes construction times in general, engineers construct 100% faster.

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
Someone pointed out how disgustingly long our construction timers are, so here's a big cull.

## Why It's Good For The Game
Hopefully encourages people to actually use one of the defining game mechanics.

## Changelog
:cl:
balance: Construction times reduced across the board
balance: Engineers & the Chief construct twice as fast.
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
I did a funny

**Comment by PowerfulBacon** [comment]
our construction timers are not that long at all, played another server where anything with an action bar felt like 5x longer than bee

**Comment by Archanial** [comment]
This pr title/body is inaccurate.

**Comment by The-Moon-Itself** [comment]
Maybe add a method to get the trait for people that didn’t join the game as engie so that job transfers and skeleton crews that need to fill the role aren’t strictly inferior.

**Comment by KubeRoot** [comment]
Why do a bunch of things have time increased from 15 to 20? Is that intended?

**Comment by Kapu1178** [comment]
> Why do a bunch of things have time increased from 15 to 20? Is that intended?

Nnnnope. Ill fix it later.

**Comment by FriendlyContractor** [comment]
Neutral on decrases across the board but i like engineers being faster thing. Nice small bonus slightly differentiating engineers from rest of the crew members and encouraging them to indulge themselves in construction

**Comment by DatBoiTim** [comment]
> 



> Maybe add a method to get the trait for people that didn’t join the game as engie so that job transfers and skeleton crews that need to fill the role aren’t strictly inferior.

#3242

**Comment by bluezorua** [comment]
> Neutral on decrases across the board but i like engineers being faster thing. Nice small bonus slightly differentiating engineers from rest of the crew members and encouraging them to indulge themselves in construction

other servers have "construction gloves" which do just this

**Comment by ToasterBan** [comment]
> Maybe add a method to get the trait for people that didn’t join the game as engie so that job transfers and skeleton crews that need to fill the role aren’t strictly inferior.

other servers have this, they are called "construction gloves", they do the same thing as the engineer trait with halving construction time

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -200,7 +200,10 @@
 				return
 			if(R.time)
 				usr.visible_message("<span class='notice'>[usr] starts building \a [R.title].</span>", "<span class='notice'>You start building \a [R.title]...</span>")
-				if (!do_after(usr, R.time, target = usr))
+				if(HAS_TRAIT(usr.mind, TRAIT_CONSTRUCTION_ADEPT))
+					if(!do_after(usr, (R.time / 2), target = usr))
+						return
+				else if(!do_after(usr, R.time, target = usr))
 					return
```
Comment (Fikou) [inline code/game/objects/items/stacks/stack.dm:207]:
```suggestion
				if(!do_after(usr, R.time * (HAS_TRAIT(usr.mind, TRAIT_CONSTRUCTION_ADEPT) ? 0.5 : 1), target = usr))
					return
```

Comment (KubeRoot) [reply code/game/objects/items/stacks/stack.dm:207]: Does DM handle precedence correctly here? Might be a good idea to wrap the ternary in parentheses if you're not 100% sure.

Comment (Kapu1178) [reply code/game/objects/items/stacks/stack.dm:207]: Tested it in game. It works fine.

**Comment by Scinexx** [comment]
This is why I use an RCD in 90% of cases, construction especially at a low server tick rate is abysmal.

**Review by PowerfulBacon — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]
This is a bit stupid but it could be used to meta if someone was an engineer at roundstart by seeing how quick they build. Perhaps it could be tied to some kind of item like gloves?
Overwise I am still for it anyway.

**Review by ike709 — CHANGES_REQUESTED** [review]
Ignoring any other balance concerns, this isn't going to merged as long as it's solely tied to a mind trait. Otherwise people can just roll Engi and then switch jobs and still have a construction bonus.

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 2]
```diff
@@ -573,18 +573,18 @@ GLOBAL_LIST_INIT(runed_metal_recipes, list ( \
...
 	null, \
-	new/datum/stack_recipe("pinion airlock", /obj/machinery/door/airlock/clockwork, 5, time=40, one_per_turf = TRUE, on_floor = TRUE), \
-	new/datum/stack_recipe("pinion windowed airlock", /obj/machinery/door/airlock/clockwork/glass, 5, time=40, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("pinion airlock", /obj/machinery/door/airlock/clockwork, 5, time=25, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("pinion windowed airlock", /obj/machinery/door/airlock/clockwork/glass, 5, time=25, one_per_turf = TRUE, on_floor = TRUE), \
 	null, \
-	new/datum/stack_recipe("brass chair", /obj/structure/chair/brass, 1, time=40, one_per_turf = TRUE, on_floor = TRUE), \
-	new/datum/stack_recipe("brass table frame", /obj/structure/table_frame/brass, 1, time=40, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("brass chair", /obj/structure/chair/brass, 1, time=5, one_per_turf = TRUE, on_floor = TRUE), \
+	new/datum/stack_recipe("brass table frame", /obj/structure/table_frame/brass, 1, time=5, one_per_turf = TRUE, on_floor = TRUE), \
```
Comment (PowerfulBacon) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:587]: Why is the brass construction time so insanely nerfed? The cogscarabs already get a 200% buildspeed modifier.

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (Archanial) [reply code/game/objects/items/stacks/stack.dm:207]: Fikou's syntax looks much clearer here

**Pull request closed without merging**